spyder/plugins/ipythonconsole.py

```diff
@@ -921,10 +937,11 @@ def execute_code(self, lines, current_client=True, clear_variables=False):
                 if not current_client:
                     # Clear console and reset namespace for
                     # dedicated clients
-                    sw.silent_execute('%clear')
                     sw.silent_execute(
                         'get_ipython().kernel.close_all_mpl_figures()')
                     sw.reset_namespace(force=True)
+                    # wait until console it's ready before executing code
+                    self.wait_shell_is_ready(sw)

```

Comment:
This has to be before the other two previous calls (i.e. `sw.silent_execute` and `sw.reset_namespace`) because those require a running kernel to work.